internal/controller/gitsync_controller.go

```diff
@@ -135,7 +135,7 @@ func (r *GitSyncReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 }
 
 // reconcile does the real logic
-func (r *GitSyncReconciler) reconcile(ctx context.Context, gitSync *apiv1.GitSync) (ctrl.Result, error) {
+func (r *GitSyncReconciler) reconcile(ctx context.Context, gitSync *apiv1.GitSync) error {
```

Comment:
removed ctrl.Result from return values since it was always set to `ctrl.Result()`